### PR TITLE
Refactor SbIcon component to SFC

### PR DESCRIPTION
**Pull request by jameslovallo**
## Pull request type

Jira Link: NA

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Other (please describe):

## How to test this PR

Please navigate to the [SbIcon page](http://localhost:6006/?path=/docs/design-system-components-sbicon--default) in Storybook and verify that the icon component behaves as expected:

- Each prop listed should have the expected effect on the rendered icon, when changed.
- For any instance of SbIcon on this page, and any other page (i.e. SbButton), the component should display the correct icon and styling.

## What is the new behavior?

- SBIcon component is refactored to SbIcon.vue.
- A new unit test, 'when pass role attribute', was added to src/components/Icon/__tests__/icon.spec.js, and the result of that test should be visible when running `yarn test:unit`

## Other information

Since this is a very simple component, I consolidated as much as possible into the .vue file. I left the utils.js file because the exported iconSizes array is used both by the .vue file and SbIcon's Storybook config file.

### CODE REVIEW

**Comment by lisilinhart** [comment]
Hi @jameslovallo, thanks for the help! We'll take a look in the next days, we need to test it with our app before we merge it.

File: src/components/Icon/SbIcon.vue [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        this.color ? `sb-icon--color-${this.color}` : null,
+        this.backgroundColor ? `sb-icon--bg-${this.backgroundColor}` : null,
+      ]
+    },
+  },
+}
+</script>
+
+<style lang="scss">
+$icon-colors: (
```
Comment (lisilinhart) [inline src/components/Icon/SbIcon.vue:69]: We tend to keep the `scss` seperated to it's own file

Comment (jameslovallo) [reply src/components/Icon/SbIcon.vue:69]: Thank you @lisilinhart. This is just for the job application process, so I'm not surprised there are duplicate PRs. I'll move scss out now.

Comment (jameslovallo) [reply src/components/Icon/SbIcon.vue]: @lisilinhart I moved the `scss` to a separate file and imported from there into the `vue` SFC. I saw that some files import the `scss` into index.js and others import it in the SFC, what is the convention for this project?

**Comment by lisilinhart** [comment]
Hi @jameslovallo, I'll close this PR because we merged the other PR that was open

**Pull request closed without merging**